test/root.test.js

```diff
@@ -45,6 +46,7 @@ test("non-empty root", () => {
     </main>
   `
 })
+*/
```

Comment:
I don't really agree, this describes special casing render target selection and is really not going to be compatible with SSR in general.  If the user is placing <script> tags in the body they should also be providing a render target that does not include those script tags imo.

The described behavior reads to me that if no target is specified and body has children, then we skip to last node and render there?  or maybe it's target first node with matching name, I didn't track down the logic covered in this test?  Either way it is some kind of special case that I think makes more sense to be an explicit choice by the user.